mito-ai/src/Extensions/AiChat/ChatTaskpane.tsx

```diff
@@ -587,11 +587,17 @@ const ChatTaskpane: React.FC<IChatTaskpaneProps> = ({
                 break;
             }
 
-            // Insert a new cell for the next step
-            await app.commands.execute("notebook:insert-cell-below")
+            // Only create a new cell if the AI generated code in its response
+            if (aiMessage?.message) {
```

Comment:
Technically this is not the most recent aiMessage anymore because at this point, we will have already gone through the retry during execution loop. But I guess this works because if the original one was empty then there is no way we would error.